cmd/node-agent/README.md

```diff
@@ -0,0 +1,59 @@
+# Node Agent
+
+## Description
+
+The Node Agent is a process that runs on each node of a Kubernetes cluster. The
+purpose of this process is to monitor a specific annotation on the node object.
+This annotation holds a JSON object which describes the version and configuration
+to be used for the on-host kubelet.
+
+This is useful for upgrading your on-host kubelet without ever having to ssh
+into a machine. Simply update the annotation with the desired version, and the
+name of the configMap that holds your configuration, and the Node Agent will
+ensure that the on-host kubelet matches those values, and will restart the on-host
+kubelet so those new changes take effect.
+
+## Usage
+
+If you are using a supported self-hosted cluster then the Node Agent will be
+running by default.
+
+The Node Agent looks for the annotation named:
+
+```
+"node-agent.alpha.coreos.com/desired-config"
+```
+
+The format of the JSON object used by the Node Agent is:
+
+```json
+{
+        "KUBELET_CONFIG": "<your configMap name>",
+        "KUBELET_VERSION": "<desired kubelet version>"
+}
+```
+
+To update the on-host kubelet, simply run a kubectl command such as:
+
+```
+$ kubectl \
+  --kubeconfig=cluster/auth/kubeconfig \
+  annotate \
+  --overwrite \
+  node <your node> \
+  node-agent.alpha.coreos.com/desired-config='{"KUBELET_VERSION": "v1.3.2_coreos.0", "KUBELET_CONFIG": "kubelet-config-v2"}'
+```
+
+You can then inspect the status of the on-host update by checking the following
+annotation value:
+
+```
+"node-agent.alpha.coreos.com/current-config"
```

Comment:
nit: would be helpful to show how to get this info. Something like:
 
`kubectl get node <node-name> -o=custom-columns=ANNOTATIONS:.metadata.annotations`